tests/components/airpatrol/conftest.py

```diff
@@ -0,0 +1,136 @@
+"""Common fixtures for the AirPatrol tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from airpatrol.api import AirPatrolAPI
+import pytest
+
+from homeassistant.components.airpatrol.const import DOMAIN
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from tests.conftest import MockConfigEntry
+
+
+@pytest.fixture
+def _mock_airpatrol_client_config_flow(get_data) -> Generator[AsyncMock, AsyncMock]:
+    """Mock an AirPatrol client and config."""
+    with (
+        patch(
+            "homeassistant.components.airpatrol.coordinator.AirPatrolAPI",
+            autospec=True,
+        ) as mock_coordinator_client,
+        patch(
+            "homeassistant.components.airpatrol.config_flow.AirPatrolAPI",
+            autospec=True,
+        ) as mock_config_flow_client,
+    ):
+        client_instance = MagicMock(spec=AirPatrolAPI)
+        client_instance.get_unique_id.return_value = "test_user_id"
+        client_instance.get_access_token.return_value = "test_access_token"
+        client_instance.get_data.return_value = get_data()
+        client_instance.set_unit_climate_data.return_value = AsyncMock()
+        mock_coordinator_client.return_value = client_instance
+        mock_coordinator_client.authenticate = AsyncMock(return_value=client_instance)
+        mock_config_flow_client.authenticate = AsyncMock(return_value=client_instance)
+        yield client_instance, mock_config_flow_client, mock_coordinator_client
```

Comment:
Just do one mock!

```suggestion
    with (
        patch(
            "homeassistant.components.airpatrol.coordinator.AirPatrolAPI",
            autospec=True,
        ) as mock_client,
        patch(
            "homeassistant.components.airpatrol.config_flow.AirPatrolAPI",
            new=mock_client,
        ),
    ):
        client_instance = mock_client.authenticate.return_value
        client_instance.get_unique_id.return_value = "test_user_id"
        client_instance.get_access_token.return_value = "test_access_token"
        client_instance.get_data.return_value = get_data()
        yield client_instance
```

Reply:
Nah, not cause of you! I tried that solution but had to alter it to the one I currently have to get it going!